### PR TITLE
Update history to persist partition config

### PR DESCRIPTION
**Pull request by Shaddoll**
**What changed?**
Update history to persist partition config in mutable state and workflow started event and forward it to matching when pushing activity and decision tasks to matching

**Why?**

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -2182,7 +2182,7 @@ func (wh *WorkflowHandler) StartWorkflowExecution(
 
 	wh.GetLogger().Debug("Start workflow execution request domainID", tag.WorkflowDomainID(domainID))
 	historyRequest, err := common.CreateHistoryStartWorkflowRequest(
-		domainID, startRequest, time.Now())
+		domainID, startRequest, time.Now(), nil)
```
Comment (davidporter-id-au) [inline service/frontend/workflowHandler.go:2185]: are you intending to add this later? 

I assume we're expecting this field to be optional and going to treat it as such?

File: service/history/historyEngine3_eventsv2_test.go [hunk 2]
```diff
@@ -252,6 +256,7 @@ func (s *engine3Suite) TestStartWorkflowExecution_BrandNew() {
 			Identity:                            identity,
 			RequestID:                           requestID,
 		},
+		PartitionConfig: partitionConfig,
```
Comment (davidporter-id-au) [inline service/history/historyEngine3_eventsv2_test.go:259]: I would not fill this in immediately unless we're sure it's going to be present in all cases.

**Review by davidporter-id-au — APPROVED** [review]

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (Shaddoll) [reply service/frontend/workflowHandler.go:2185]: it will be added later

File: service/history/historyEngine3_eventsv2_test.go (hunk 2 above)
Comment (Shaddoll) [reply service/history/historyEngine3_eventsv2_test.go:259]: the change is backward compatible, so previous tests covered the case when it's nil

**Pull request merged**